http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeServiceDbQueries.scala

```diff
@@ -409,4 +639,50 @@ object RuntimeServiceDbQueries {
 
     sqlStatement.as[ListRuntimeResponse2]
   }
+
+  /**
+   * Build the filter expression which shows only runtimes the user has permission to read. If any of the parameter
+   * lists is blank, the user is assumed to have no permissions (can see nothing).
+   * @param readerRuntimeIds List[String] runtime IDs the user has any Sam role for
+   * @param readerWorkspaceIds List[String] workspace IDs the user has any Sam role for
+   * @param ownerWorkspaceIds List[String] Google project IDs the user has an owner role for
+   * @param readerGoogleProjectIds List[String] Google project IDs the user has any Sam role for
+   * @param ownerGoogleProjectIds List[String] Google project IDs the user has an owner role for
+   * @return List[String] a one-element list with the SQL filter expression
+   */
+  private def getFilterVisibleIds(readerRuntimeIds: List[String],
```

Comment:
It'd be nice to use custom types instead of `String`s for all these IDs

Reply:
done for the point of entry (`listAuthorizedRuntimes`). For these private methods I don't think its as needed, if that seems ok!